### ISSUE TITLE
what difference should I see depending on format in list_packages()?

### ISSUE THREAD

**Issue by maelle** [r-hub/pkgsearch#43]
``` r
library("pkgsearch")
list_packages(format = "short")
#> CRAN packages (short)-----------------------------------------------------
#>  Package     Version Title                                                
#>  A3          1.0.0   Accurate, Adaptable, and Accessible Error Metrics ...
#>  abbyyR      0.5.4   Access to Abbyy Optical Character Recognition (OCR...
#>  abc         2.1     Tools for Approximate Bayesian Computation (ABC)     
#>  abc.data    1.0     Data Only: Tools for Approximate Bayesian Computat...
#>  ABC.RAP     0.9.0   Array Based CpG Region Analysis Pipeline             
#>  ABCanalysis 1.2.1   Computed ABC Analysis                                
#>  abcdeFBA    0.4     ABCDE_FBA: A-Biologist-Can-Do-Everything of Flux B...
#>  ABCoptim    0.15.0  Implementation of Artificial Bee Colony (ABC) Opti...
#>  ABCp2       1.2     Approximate Bayesian Computational Model for Estim...
#>  abcrf       1.7.1   Approximate Bayesian Computation via Random Forest...
list_packages(format = "latest")
#> CRAN packages (latest versions)-------------------------------------------
#>  Package     Version Title                                                
#>  A3          1.0.0   Accurate, Adaptable, and Accessible Error Metrics ...
#>  abbyyR      0.5.4   Access to Abbyy Optical Character Recognition (OCR...
#>  abc         2.1     Tools for Approximate Bayesian Computation (ABC)     
#>  abc.data    1.0     Data Only: Tools for Approximate Bayesian Computat...
#>  ABC.RAP     0.9.0   Array Based CpG Region Analysis Pipeline             
#>  ABCanalysis 1.2.1   Computed ABC Analysis                                
#>  abcdeFBA    0.4     ABCDE_FBA: A-Biologist-Can-Do-Everything of Flux B...
#>  ABCoptim    0.15.0  Implementation of Artificial Bee Colony (ABC) Opti...
#>  ABCp2       1.2     Approximate Bayesian Computational Model for Estim...
#>  abcrf       1.7.1   Approximate Bayesian Computation via Random Forest...
list_packages(format = "full")
#> CRAN packages (full)------------------------------------------------------
#>  Package     Version Title                                                
#>  A3          1.0.0   Accurate, Adaptable, and Accessible Error Metrics ...
#>  abbyyR      0.5.4   Access to Abbyy Optical Character Recognition (OCR...
#>  abc         2.1     Tools for Approximate Bayesian Computation (ABC)     
#>  abc.data    1.0     Data Only: Tools for Approximate Bayesian Computat...
#>  ABC.RAP     0.9.0   Array Based CpG Region Analysis Pipeline             
#>  ABCanalysis 1.2.1   Computed ABC Analysis                                
#>  abcdeFBA    0.4     ABCDE_FBA: A-Biologist-Can-Do-Everything of Flux B...
#>  ABCoptim    0.15.0  Implementation of Artificial Bee Colony (ABC) Opti...
#>  ABCp2       1.2     Approximate Bayesian Computational Model for Estim...
#>  abcrf       1.7.1   Approximate Bayesian Computation via Random Forest...
```

<sup>Created on 2019-05-13 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

<details>

<summary>Session info</summary>

``` r
devtools::session_info()
#> ─ Session info ──────────────────────────────────────────────────────────
#>  setting  value                       
#>  version  R version 3.5.3 (2019-03-11)
#>  os       Ubuntu 18.04.2 LTS          
#>  system   x86_64, linux-gnu           
#>  ui       X11                         
#>  language en_US                       
#>  collate  en_US.UTF-8                 
#>  ctype    en_US.UTF-8                 
#>  tz       Europe/Paris                
#>  date     2019-05-13                  
#> 
#> ─ Packages ──────────────────────────────────────────────────────────────
#>  package     * version    date       lib source                         
#>  assertthat    0.2.1      2019-03-21 [1] CRAN (R 3.5.3)                 
#>  backports     1.1.4      2019-04-10 [1] CRAN (R 3.5.3)                 
#>  callr         3.2.0      2019-03-15 [1] CRAN (R 3.5.3)                 
#>  cli           1.1.0      2019-03-19 [1] CRAN (R 3.5.3)                 
#>  crayon        1.3.4      2017-09-16 [1] CRAN (R 3.5.3)                 
#>  curl          3.3        2019-01-10 [1] CRAN (R 3.5.3)                 
#>  desc          1.2.0      2019-05-07 [1] Github (r-lib/desc@c860e7b)    
#>  devtools      2.0.2      2019-04-08 [1] CRAN (R 3.5.3)                 
#>  digest        0.6.18     2018-10-10 [1] CRAN (R 3.5.3)                 
#>  evaluate      0.13       2019-02-12 [1] CRAN (R 3.5.3)                 
#>  fs            1.2.7      2019-03-19 [1] CRAN (R 3.5.3)                 
#>  glue          1.3.1.9000 2019-05-06 [1] Github (tidyverse/glue@ea0edcb)
#>  highr         0.8        2019-03-20 [1] CRAN (R 3.5.3)                 
#>  htmltools     0.3.6      2017-04-28 [1] CRAN (R 3.5.3)                 
#>  httr          1.4.0      2018-12-11 [1] CRAN (R 3.5.3)                 
#>  jsonlite      1.6        2018-12-07 [1] CRAN (R 3.5.3)                 
#>  knitr         1.22       2019-03-08 [1] CRAN (R 3.5.3)                 
#>  magrittr      1.5        2014-11-22 [1] CRAN (R 3.5.3)                 
#>  memoise       1.1.0      2017-04-21 [1] CRAN (R 3.5.3)                 
#>  parsedate     1.1.3      2017-03-02 [1] CRAN (R 3.5.3)                 
#>  pkgbuild      1.0.3      2019-03-20 [1] CRAN (R 3.5.3)                 
#>  pkgload       1.0.2      2018-10-29 [1] CRAN (R 3.5.3)                 
#>  pkgsearch   * 2.0.1.9000 2019-05-13 [1] local                          
#>  prettyunits   1.0.2      2015-07-13 [1] CRAN (R 3.5.3)                 
#>  processx      3.3.0      2019-03-10 [1] CRAN (R 3.5.3)                 
#>  ps            1.3.0      2018-12-21 [1] CRAN (R 3.5.3)                 
#>  R6            2.4.0      2019-02-14 [1] CRAN (R 3.5.3)                 
#>  Rcpp          1.0.1      2019-03-17 [1] CRAN (R 3.5.3)                 
#>  remotes       2.0.4      2019-04-10 [1] CRAN (R 3.5.3)                 
#>  rlang         0.3.4      2019-04-07 [1] CRAN (R 3.5.3)                 
#>  rmarkdown     1.12       2019-03-14 [1] CRAN (R 3.5.3)                 
#>  rprojroot     1.3-2      2018-01-03 [1] CRAN (R 3.5.3)                 
#>  sessioninfo   1.1.1      2018-11-05 [1] CRAN (R 3.5.3)                 
#>  stringi       1.4.3      2019-03-12 [1] CRAN (R 3.5.3)                 
#>  stringr       1.4.0      2019-02-10 [1] CRAN (R 3.5.3)                 
#>  testthat      2.1.1      2019-04-23 [1] CRAN (R 3.5.3)                 
#>  usethis       1.5.0      2019-04-07 [1] CRAN (R 3.5.3)                 
#>  withr         2.1.2      2018-03-15 [1] CRAN (R 3.5.3)                 
#>  xfun          0.6        2019-04-02 [1] CRAN (R 3.5.3)                 
#>  yaml          2.2.0      2018-07-25 [1] CRAN (R 3.5.3)                 
#> 
#> [1] /home/maelle/R/x86_64-pc-linux-gnu-library/3.5
#> [2] /usr/local/lib/R/site-library
#> [3] /usr/lib/R/site-library
#> [4] /usr/lib/R/library
```

</details>

**Comment by maelle** [comment]
The output is actually different, but the print method is the same.

**Comment by maelle** [comment]
@gaborcsardi should the print methods be different and if so, how should the printed thing look like for each format?

Currently no matter what the packages list actually contains, the method is

```r
print_cran_list_i <- function(x, header, version, title, ...) {
  cat_fill(header)
  pkgs <- data.frame(
    stringsAsFactors = FALSE,
    Package = names(x),
    Version = sapply(x, function(xx) xx[[version]] %>% NULL_NA),
    RTitle = sapply(x, function(xx) xx[[title]] %>% NULL_NA) %>%
      gsub(pattern = "\\s+", replacement = " ")
  )
  
  tw <- getOption("width") - 7 -
    max(nchar("Package"), max(nchar(pkgs$Package))) -
    max(nchar("Version"), max(nchar(pkgs$Version)))
  pkgs$Title <- substring(pkgs$RTitle, 1, tw)
  pkgs$Title <- ifelse(pkgs$Title == pkgs$RTitle, pkgs$Title,
                       paste0(pkgs$Title, "..."))
  pkgs$RTitle <- NULL
  
  print.data.frame(pkgs, row.names = FALSE, right = FALSE)
}
```

**Comment by gaborcsardi** [comment]
Don't worry too much about this, you can also just remove one format.

**Comment by maelle** [comment]
why would we remove them? The output has different info, it's just printed the same way in the console.

**Comment by gaborcsardi** [comment]
Why not just keep the one with the more info?

**Comment by maelle** [comment]
ok good idea, will make a PR :-) does this mean you'd get rid of the other endpoints in the API, keeping only allall and all?

**Comment by gaborcsardi** [comment]
Oh, wait, these use different end points? Then we can keep them all, just rename the `format` argument to sg more meaningful? Or create three different functions for them?

**Comment by maelle** [comment]
yes they use different endpoints.

I don't think format is a bad name, but printing gives the impression nothing happens.

**Comment by maelle** [comment]
In the docs I could use this example (that'd be rendered in the pkgdown man pages, and in the README)

``` r
library("pkgsearch")
l1 <- list_packages(format = "short")
l2 <- list_packages(format = "latest")
l3 <- list_packages(format = "full")

l1[[1]]
#> $version
#> [1] "1.0.0"
#> 
#> $title
#> [1] "Accurate, Adaptable, and Accessible Error Metrics for Predictive\nModels"
l2[[1]]
#> $Package
#> [1] "A3"
#> 
#> $Type
#> [1] "Package"
#> 
#> $Title
#> [1] "Accurate, Adaptable, and Accessible Error Metrics for Predictive\nModels"
#> 
#> $Version
#> [1] "1.0.0"
#> 
#> $Date
#> [1] "2015-08-15"
#> 
#> $Author
#> [1] "Scott Fortmann-Roe"
#> 
#> $Maintainer
#> [1] "Scott Fortmann-Roe <scottfr@berkeley.edu>"
#> 
#> $Description
#> [1] "Supplies tools for tabulating and analyzing the results of predictive models. The methods employed are applicable to virtually any predictive model and make comparisons between different methodologies straightforward."
#> 
#> $License
#> [1] "GPL (>= 2)"
#> 
#> $Depends
#> $Depends$R
#> [1] ">= 2.15.0"
#> 
#> $Depends$xtable
#> [1] "*"
#> 
#> $Depends$pbapply
#> [1] "*"
#> 
#> 
#> $Suggests
#> $Suggests$randomForest
#> [1] "*"
#> 
#> $Suggests$e1071
#> [1] "*"
#> 
#> 
#> $NeedsCompilation
#> [1] "no"
#> 
#> $Packaged
#> [1] "2015-08-16 14:17:33 UTC; scott"
#> 
#> $Repository
#> [1] "CRAN"
#> 
#> $`Date/Publication`
#> [1] "2015-08-16 23:05:52"
#> 
#> $crandb_file_date
#> [1] "2015-08-16 17:08:22"
#> 
#> $date
#> [1] "2015-08-16T23:05:52+00:00"
#> 
#> $releases
#> list()
l3[[1]]
#> $name
#> [1] "A3"
#> 
#> $versions
#> $versions$`0.9.1`
#> $versions$`0.9.1`$Package
#> [1] "A3"
#> 
#> $versions$`0.9.1`$Type
#> [1] "Package"
#> 
#> $versions$`0.9.1`$Title
#> [1] "A3: Accurate, Adaptable, and Accessible Error Metrics for<U+000a>Predictive Models"
#> 
#> $versions$`0.9.1`$Version
#> [1] "0.9.1"
#> 
#> $versions$`0.9.1`$Date
#> [1] "2013-02-06"
#> 
#> $versions$`0.9.1`$Author
#> [1] "Scott Fortmann-Roe"
#> 
#> $versions$`0.9.1`$Maintainer
#> [1] "Scott Fortmann-Roe <scottfr@berkeley.edu>"
#> 
#> $versions$`0.9.1`$Description
#> [1] "This package supplies tools for tabulating and analyzing<U+000a>the results of predictive models. The methods employed are<U+000a>applicable to virtually any predictive model and make<U+000a>comparisons between different methodologies straightforward."
#> 
#> $versions$`0.9.1`$License
#> [1] "GPL (>= 2)"
#> 
#> $versions$`0.9.1`$Depends
#> $versions$`0.9.1`$Depends$xtable
#> [1] "*"
#> 
#> $versions$`0.9.1`$Depends$pbapply
#> [1] "*"
#> 
#> 
#> $versions$`0.9.1`$Suggests
#> $versions$`0.9.1`$Suggests$randomForest
#> [1] "*"
#> 
#> $versions$`0.9.1`$Suggests$e1071
#> [1] "*"
#> 
#> 
#> $versions$`0.9.1`$Packaged
#> [1] "2013-02-06 16:46:12 UTC; scott"
#> 
#> $versions$`0.9.1`$Repository
#> [1] "CRAN"
#> 
#> $versions$`0.9.1`$`Date/Publication`
#> [1] "2013-02-07 10:00:27"
#> 
#> $versions$`0.9.1`$crandb_file_date
#> [1] "2013-02-07 04:00:29"
#> 
#> $versions$`0.9.1`$date
#> [1] "2013-02-07T10:00:27+00:00"
#> 
#> $versions$`0.9.1`$releases
#> [1] "2.15.3"
#> 
#> 
#> $versions$`0.9.2`
#> $versions$`0.9.2`$Package
#> [1] "A3"
#> 
#> $versions$`0.9.2`$Type
#> [1] "Package"
#> 
#> $versions$`0.9.2`$Title
#> [1] "A3: Accurate, Adaptable, and Accessible Error Metrics for<U+000a>Predictive Models"
#> 
#> $versions$`0.9.2`$Version
#> [1] "0.9.2"
#> 
#> $versions$`0.9.2`$Date
#> [1] "2013-03-24"
#> 
#> $versions$`0.9.2`$Author
#> [1] "Scott Fortmann-Roe"
#> 
#> $versions$`0.9.2`$Maintainer
#> [1] "Scott Fortmann-Roe <scottfr@berkeley.edu>"
#> 
#> $versions$`0.9.2`$Description
#> [1] "This package supplies tools for tabulating and analyzing<U+000a>the results of predictive models. The methods employed are<U+000a>applicable to virtually any predictive model and make<U+000a>comparisons between different methodologies straightforward."
#> 
#> $versions$`0.9.2`$License
#> [1] "GPL (>= 2)"
#> 
#> $versions$`0.9.2`$Depends
#> $versions$`0.9.2`$Depends$R
#> [1] ">= 2.15.0"
#> 
#> $versions$`0.9.2`$Depends$xtable
#> [1] "*"
#> 
#> $versions$`0.9.2`$Depends$pbapply
#> [1] "*"
#> 
#> 
#> $versions$`0.9.2`$Suggests
#> $versions$`0.9.2`$Suggests$randomForest
#> [1] "*"
#> 
#> $versions$`0.9.2`$Suggests$e1071
#> [1] "*"
#> 
#> 
#> $versions$`0.9.2`$Packaged
#> [1] "2013-03-26 18:58:12 UTC; scott"
#> 
#> $versions$`0.9.2`$Repository
#> [1] "CRAN"
#> 
#> $versions$`0.9.2`$`Date/Publication`
#> [1] "2013-03-26 19:58:40"
#> 
#> $versions$`0.9.2`$crandb_file_date
#> [1] "2013-03-26 14:58:40"
#> 
#> $versions$`0.9.2`$NeedsCompilation
#> [1] "no"
#> 
#> $versions$`0.9.2`$date
#> [1] "2013-03-26T19:58:40+00:00"
#> 
#> $versions$`0.9.2`$releases
#> [1] "3.0.0" "3.0.1" "3.0.2" "3.0.3" "3.1.0" "3.1.1"
#> 
#> 
#> $versions$`1.0.0`
#> $versions$`1.0.0`$Package
#> [1] "A3"
#> 
#> $versions$`1.0.0`$Type
#> [1] "Package"
#> 
#> $versions$`1.0.0`$Title
#> [1] "Accurate, Adaptable, and Accessible Error Metrics for Predictive\nModels"
#> 
#> $versions$`1.0.0`$Version
#> [1] "1.0.0"
#> 
#> $versions$`1.0.0`$Date
#> [1] "2015-08-15"
#> 
#> $versions$`1.0.0`$Author
#> [1] "Scott Fortmann-Roe"
#> 
#> $versions$`1.0.0`$Maintainer
#> [1] "Scott Fortmann-Roe <scottfr@berkeley.edu>"
#> 
#> $versions$`1.0.0`$Description
#> [1] "Supplies tools for tabulating and analyzing the results of predictive models. The methods employed are applicable to virtually any predictive model and make comparisons between different methodologies straightforward."
#> 
#> $versions$`1.0.0`$License
#> [1] "GPL (>= 2)"
#> 
#> $versions$`1.0.0`$Depends
#> $versions$`1.0.0`$Depends$R
#> [1] ">= 2.15.0"
#> 
#> $versions$`1.0.0`$Depends$xtable
#> [1] "*"
#> 
#> $versions$`1.0.0`$Depends$pbapply
#> [1] "*"
#> 
#> 
#> $versions$`1.0.0`$Suggests
#> $versions$`1.0.0`$Suggests$randomForest
#> [1] "*"
#> 
#> $versions$`1.0.0`$Suggests$e1071
#> [1] "*"
#> 
#> 
#> $versions$`1.0.0`$NeedsCompilation
#> [1] "no"
#> 
#> $versions$`1.0.0`$Packaged
#> [1] "2015-08-16 14:17:33 UTC; scott"
#> 
#> $versions$`1.0.0`$Repository
#> [1] "CRAN"
#> 
#> $versions$`1.0.0`$`Date/Publication`
#> [1] "2015-08-16 23:05:52"
#> 
#> $versions$`1.0.0`$crandb_file_date
#> [1] "2015-08-16 17:08:22"
#> 
#> $versions$`1.0.0`$date
#> [1] "2015-08-16T23:05:52+00:00"
#> 
#> $versions$`1.0.0`$releases
#> list()
#> 
#> 
#> 
#> $timeline
#> $timeline$`0.9.1`
#> [1] "2013-02-07T10:00:27+00:00"
#> 
#> $timeline$`0.9.2`
#> [1] "2013-03-26T19:58:40+00:00"
#> 
#> $timeline$`1.0.0`
#> [1] "2015-08-16T23:05:52+00:00"
#> 
#> 
#> $latest
#> [1] "1.0.0"
#> 
#> $title
#> [1] "Accurate, Adaptable, and Accessible Error Metrics for Predictive\nModels"
#> 
#> $archived
#> [1] FALSE
```

<sup>Created on 2019-05-13 by the [reprex package](https://reprex.tidyverse.org) (v0.2.1)</sup>

**Comment by maelle** [comment]
@gaborcsardi I updated #45

**Comment by maelle** [comment]
In https://github.com/r-hub/pkgsearch/pull/45#discussion_r283356114 @gaborcsardi  said that we should just return tibbles.

**Comment by maelle** [comment]
There will be a need for choices when designing said tibbles from the nested lists.

**Comment by maelle** [comment]
fixed by #47
